Frameworks/ImageIO/CGImageDestination.mm

```diff
@@ -15,11 +16,847 @@
+        }
+
+        setVariantFromDictionary(tiffDictionary,
+                                 kCGImagePropertyTIFFSoftware,
+                                 VT_LPSTR,
+                                 L"/ifd/{ushort=305}",
+                                 propertyWriter);
+
+        setVariantFromDictionary(tiffDictionary,
+                                 kCGImagePropertyTIFFDateTime,
+                                 VT_LPSTR,
+                                 L"/ifd/{ushort=306}",
+                                 propertyWriter);
+
+        setVariantFromDictionary(tiffDictionary,
+                                 kCGImagePropertyTIFFArtist,
+                                 VT_LPSTR,
+                                 L"/ifd/{ushort=315}",
+                                 propertyWriter);
+
+        setVariantFromDictionary(tiffDictionary,
+                                 kCGImagePropertyTIFFCopyright,
+                                 VT_LPSTR,
+                                 L"/ifd/{ushort=33432}",
+                                 propertyWriter);
+
+        setVariantFromDictionary(tiffDictionary,
+                                 kCGImagePropertyTIFFWhitePoint,
+                                 VT_UI2,
+                                 L"/ifd/{ushort=41987}",
+                                 propertyWriter);
+    }
+}
+
+void writePNGProperties(IWICMetadataQueryWriter* propertyWriter, CFDictionaryRef properties, int imageWidth, int imageHeight) {
+    PROPVARIANT propertyToWrite;
+
+    // General note for PNG: Apple writes a lot of properties for PNG that are not defined in the w3 PNG specifications.
+    // This includes the entire GPS dictionary, Exif information, and DPI information. Apple APIs will read these back
+    // properly, but external metadata readers don't seem to pick up on the information and the location of this metadata is
+    // not well specified. This behavior is not replicated here.
+
+    setVariantFromDictionary(properties,
+                             kCGImagePropertyProfileName,
+                             VT_LPSTR,
+                             L"/iCCP/ProfileName",
+                             propertyWriter);
+
+    if (properties && CFDictionaryContainsKey(properties, kCGImagePropertyPNGDictionary)) {
+        CFDictionaryRef pngDictionary = (CFDictionaryRef)CFDictionaryGetValue(properties, kCGImagePropertyPNGDictionary);
+
+        setVariantFromDictionary(pngDictionary,
+                                 kCGImagePropertyPNGGamma,
+                                 VT_UI4,
+                                 L"/gAMA/ImageGamma",
+                                 propertyWriter);
+
+        setVariantFromDictionary(pngDictionary,
+                                 kCGImagePropertyPNGsRGBIntent,
+                                 VT_UI1,
+                                 L"/sRGB/RenderingIntent",
+                                 propertyWriter);
+    }
```

Comment:
I'm not sure there's any reason to because there are other properties not being implemented and I don't think it's reasonable to have unimplemented tracing for every single property we don't use.